extensions/virtual-threads/runtime/src/main/java/io/quarkus/virtual/threads/VirtualThreadsRecorder.java

```diff
@@ -20,7 +20,7 @@ public class VirtualThreadsRecorder {
 
     private static final Logger logger = Logger.getLogger("io.quarkus.virtual-threads");
 
-    static VirtualThreadsConfig config = new VirtualThreadsConfig();
+    static volatile VirtualThreadsConfig config;
 
```

Comment:
Is there anything specific here that prevents us from using injection in build steps?

In the test below we could pass the config as a parameter?

(sorry on my phone in a waiting room…)

Reply:
I kept the original code because I didn't want to break anything, but maybe I can improve that

Reply:
Yeah fine with me, I was just surprised. We had quite a lot of bad surprises with this pattern.

Reply:
It doesn't look like it's avoidable. It is necessary to create a new executor (`io.quarkus.virtual.threads.VirtualThreadsRecorder#createExecutor`) and I don't see a better way to do that without looking up the object again